React/Views/RCTModalHostViewManager.m

```diff
@@ -92,7 +98,13 @@ - (void)dismissModalHostView:(RCTModalHostView *)modalHostView
   if (_dismissalBlock) {
     _dismissalBlock([modalHostView reactViewController], viewController, animated, nil);
   } else {
-    [viewController.presentingViewController dismissViewControllerAnimated:animated completion:nil];
+    self.dismissWaitingBlock = ^{
+      [viewController.presentingViewController dismissViewControllerAnimated:animated completion:nil];
```

Comment:
one idea:

Could we just call `[viewController dismissViewControllerAnimated:completion:]` instead of using `presentingViewController`? 

I can't test it myself since I can't reproduce it.